aws/lambdas/hhvm-get-build-status/index.js

```diff
@@ -63,30 +69,57 @@ exports.handler = async (event) => {
 
   let success = true;
   let results = {};
+  let unpublished = {};
   const s3 = new AWS.S3();
   await Promise.all(
     Object.values(paths).map(async path => {
       try {
         await s3.headObject(
           { Bucket: 'hhvm-downloads', Key: path }
         ).promise();
-        results[path] = true;
+        results[path] = 'success';
       } catch (err) {
-        results[path] = false;
         success = false;
+        results[path] = 'failure';
+        if (unpublished_paths[path]) {
+          try {
+            await s3.headObject(
+              { Bucket: 'hhvm-scratch', Key: unpublished_paths[path] }
```

Comment:
Since this check is only in the failure branch, it won't catch the case where the package is uploaded but then the publishing step fails to update the necessary index/manifest/etc., or the Docker images are not generated, or whatever else we do there. Should we handle that case ("partially published") as well?

Reply:
That would be ideal, but a substantial increase in the scope of this; it would need to support querying dockerhub, and parsing the metadata files. Not against it, but IMO doens't belong in this diff

Reply:
Of course, it doesn't have to be in this diff.

But I was thinking maybe we could just have this same exact check in the success branch -- i.e., if the package exists in the public S3 but it also still exists in hhvm-scratch, then we know the publish script didn't finish (it didn't reach the point where it deletes packages from scratch), so we could consider that state "partially published".